src/app/Http/Controllers/SwitcherController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\ServiceBodyResource;
+use App\Http\Resources\ServiceBodyResourceJsonp;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Legacy\LegacyController;
+use App\Interfaces\ServiceBodyRepositoryInterface;
+
+class SwitcherController extends Controller
+{
+    private ServiceBodyRepositoryInterface $serviceBodyRepository;
+
+    public function __construct(ServiceBodyRepositoryInterface $serviceBodyRepository)
+    {
+        $this->serviceBodyRepository = $serviceBodyRepository;
+    }
+
+    public function get(Request $request, string $dataFormat)
+    {
+        $isJsonp = $dataFormat == 'jsonp';
+        $callback = $request->input('callback', 'callback');
+        $switcher = $request->input('switcher');
+
+        if (in_array($switcher, ['GetServiceBodies'])) {
+            $response = null;
+
+            if ($switcher == 'GetServiceBodies') {
+                $serviceBodies = $this->getServiceBodies($request);
+                if (!$serviceBodies->exists()) {
+                    // TODO look at clients and see if this is really necessary... I would much rather
+                    // TODO just return an empty list
```

Comment:
I can check on this, I wouldn't be surprised if some of them don't even check for empty service bodies in a server

Reply:
just checked, bread would not be affected from api if it were an empty list

Reply:
it doesnt appear yap would be either

Reply:
note that they would have separate issues with a server with no service bodies but nothing to do with empty response changing to empty list

Reply:
I think I recall some client checking for {} from GetSearchResults. Just couldn't remember.

Reply:
ah ok I specifically just checked GetServiceBodies, I think your right about GetSearchResults, ill see what client does that

Reply:
yes both yap and crouton check for `{}` in GetSearchResults

Reply:
Nice, we should open PRs to both of them to work with both {} and [].

Reply:
and then what put a requirement on 3.0 requiring certain versions of yap/crouton

Reply:
yes

Reply:
Cool

Reply:
Is this still relevant https://github.com/bmlt-enabled/bmlt-root-server/issues/12

Reply:
Yes, that's what we're talking about. We should fix the clients to to work with an object or a list for now, because 3.0 is going to only return lists.